### ISSUE TITLE
MD044: values in 'names' configuration are flagged in inline HTML

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#443]
### rumdl version

0.1.23

### Installation method

mise

### Bug description

URLs in inline HTML are always all lowercase.

Values can be set in `names` configuration in MD044 that are not all lowercase but can match part of a URL.

The MD044 check will see these matches as violations rather than ignoring them because they are parts of a URL in inline HTML.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   # Heading

   test

   <img src="www.example.test/test_image.png">
   ```
1. Create a `.rumdl.toml` file with this content:
   ```toml
   [MD033]
   allowed-elements = ["img"]

   [MD044]
   names = ["Test"]
   ```
1. Run `rumdl check` on `test.md`.

### Expected behavior

`rumdl` would flag only line 3.

### Actual behavior

`rumdl` also flags line 5 twice.

### Configuration

```toml
[MD033]
allowed-elements = ["img"]

[MD044]
names = ["Test"]
```

### Operating system

macOS

**Comment by rvben** [comment]
Thanks for the report, @eread\! Turns out attribute values aren't really prose, so flagging proper names inside them doesn't make sense. Fixed on main and will ship in the next release. Text between HTML tags (like anchor text) is still checked as usual.

**Comment by eread** [comment]
> Thanks for the report, [@eread](https://github.com/eread)! Turns out attribute values aren't really prose, so flagging proper names inside them doesn't make sense. Fixed on main and will ship in the next release. Text between HTML tags (like anchor text) is still checked as usual.

Thanks @rvben! I notice in the release notes: https://github.com/rvben/rumdl/releases/tag/v0.1.24, you mention that the fix only applies to false positives that are adjacent to underscores. Wouldn't you want the fix to apply to more examples?

For example, in the sample I give above `<img src="www.example.test/test_image.png">`, with the fix in place, the second instance of `test` (in `test_image`) is no longer flagged, but the first instance of `test` (in `.test`) still is.

Wouldn't you want everything between the double quotes in `src=""` to be excluded?

**Comment by rvben** [comment]
Only a part of the fix was part of the actual release, the full fix came in a commit after the release. All things inside html tags will be ignored from next release onwards.

**Comment by rvben** [comment]
Now it's released too, in 0.1.25. @eread, let me know if you hit any issue.

**Comment by eread** [comment]
> Now it's released too, in 0.1.25. [@eread](https://github.com/eread), let me know if you hit any issue.

Thanks @rvben. Looks like it's working well!
